lms/djangoapps/commerce/api/v0/views.py

```diff
@@ -194,6 +192,17 @@ def post(self, request, *args, **kwargs):
         self._handle_marketing_opt_in(request, course_key, user)
         return response
 
+    def _add_request_cookie_to_api_session(self, server_session, request, cookie_name):
+        user_cookie = None
+        if cookie_name:
+            user_cookie = request.COOKIES.get(cookie_name)
+        if user_cookie:
+            server_cookie = {cookie_name: user_cookie}
+            if server_session.cookies:
```

Comment:
the first check is to make sure that cookie_name isn't null, which is the case if the user doesn't have a cookie for UTM or sailthru